helper/debug.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+const _ = require('lodash');
+
+class Debug {
+  constructor(moduleName){
+    this.name = moduleName || 'unnamed module';
+  }
+
+  push(req, debugMsg){
+    if (!_.isEmpty(req.clean) && req.clean.enableDebug){
+      req.debug = req.debug || [];
+      // remove the extra space character
+      req.debug.push({[this.name]: debugMsg});
+    //  req.debug.push(`[${this.name}] ${debugMsg}`);
+    }
+  }
```

Comment:
It would be great if there was 0 performance penalty when debugging was disabled.

As it is, there is a need to perform 'work' on the message to compose and format the error string, this is also true of objects and may result in timely code and external function calls which end up being needless if debugging is enabled.

for example:
```javascript
debug.push( req, foo.getName() + " timer took " + ( baz / Math.round( bar ) ) )
```

In this case ^ the production environment will pay the cost of the function call `getName()` plus the math functions when in the end, none of it is returned to the user.

A nice clean way of minimizing the performance penalty when debugging is turned off, is to only allow users to pass **functions** to `push()`

for example:

```javascript
debug.push( req, function(){
   // do some work, but only if debugging is enabled
});
```

or if you decide to pass `req` in the constructor:

```javascript
debug.push( function( req ){
   return foo.getName() + " timer took " + ( baz / Math.round( bar ) )
});
```

there would be a minor cost associated with creating the function, but the time taken to compose and format the message would only be paid when debugging was enabled, not when it's disabled (in production).

another thing to consider is how timers are used, if another debug message is defined while a timer is running, then the cost of the debugging code will be included in the timer, which will make the timers far less intuitive when tracking down slow code paths.